get_goes_x.py

```diff
@@ -31,33 +33,32 @@
     print(('Warning: failed to open URL {}: {}'.format(url, last_err)))
     sys.exit(0)
 
-
-colnames = 'year month dom  hhmm  mjd secs short long ratio'.split()
-data_colnames = colnames[-3:]
-
 try:
-    dat = ascii.read(urldat, guess=False, Reader=ascii.NoHeader,
-                     data_start=3, names=colnames)
+    dat = Table(json.loads(urldat))
 except Exception as err:
     print(('Warning: malformed GOES_X data so table read failed: {}'
           .format(err)))
     sys.exit(0)
 
-# Strip up to two rows at the end if any values are bad (i.e. negative)
-for _ in range(2):
-    if any(dat[name][-1] < 0 for name in data_colnames):
-        dat = dat[:-1]
-
-mjd = dat['mjd'] + dat['secs'] / 86400.
+# Select only the GOES 16 data (which is all there is)
+dat = dat[dat['satellite'] == 16]
+# Make a table for each of the two wavelengths
+shortdat = dat[dat['energy'] == '0.05-0.4nm']['flux', 'time_tag']
+shortdat.rename_column('flux', 'short')
+longdat = dat[dat['energy'] == '0.1-0.8nm']['flux', 'time_tag']
+longdat.rename_column('flux', 'long')
+# Join them on time (seems to be OK for these data)
```

Comment:
Can you add a warning if the two `time_tag` columns are not exactly equal?  With the `join` below you can silently be dropping data and never know.

Reply:
That's a good idea, though without more information about the detectors I was inferring that the instrument just gives these two measurements all the time and the json file with separate records for them was just there to irritate me.  What are we going to do with an email warning about, say, missing data in one channel excluding data in the other channel?

Reply:
We would be informed that data are being dropped from the file and would evaluate how much data are being dropped and whether a fix is needed, for instance interpolating across missing points.